src/coreclr/tools/ILVerification/AccessVerificationHelpers.cs

```diff
@@ -204,11 +187,21 @@ private static bool CanAccessMethodSignature(this TypeDesc currentType, MethodDe
                     return false;
             }
 
-            return currentType.CanAccessSignatureArguments(methodSig);
+            return currentType.CanAccessSignature(methodSig);
         }
 
-        private static bool CanAccessSignatureArguments(this TypeDesc currentType, MethodSignature signature)
+        private static bool CanAccessSignature(this TypeDesc currentType, MethodSignature signature)
         {
+            // Check return type
+            var returnType = signature.ReturnType;
+
+            if (returnType.IsByRef)
```

Comment:
These checks seems to be redundant with what is done by `CanAccess` already. It may be better to simplify this to just:

```
    if (!currentType.CanAccess(signature.ReturnType))
        return false;
```

Similarly, the checks of individual arguments below can be just `if (!currentType.CanAccess(signature[i])) ...`.